src/NServiceBus.TransactionalSession/TransactionalSessionDelayControlMessageBehavior.cs

```diff
@@ -31,23 +31,25 @@ public async Task Invoke(IIncomingPhysicalMessageContext context,
         var messageId = context.MessageId;
         if (remainingCommitDuration <= TimeSpan.Zero)
         {
-            if (Log.IsInfoEnabled)
+            if (Log.IsWarnEnabled)
```

Comment:
@lailabougria and @tmasternak I know you have also been involved with the transactional session design. As far as I remember, we made this deliberately an INFO log level because we are considering the tombstoning path an expected or normal outcome independent of the setting.

What is the rationale for bumping to log warn for something that is an expected outcome? I get that from the angle of "users probably only have log warn enabled in production" the WARN might make it more visible in case they have indeed misconfigured it. But for the real tombstoning cases, for example when the API call errorred, this would raise warnings for something the user can absolutely do nothing.

By the way the new improved message is "chef kiss"

Reply:
Or maybe to put it from a different angle. If we think there is a certain threshold when this should be treated as a warning we should probably introduce some kind of "circuit breaker" with configuration that flips the switch and starts logging it as warning and flips back once things settled.